support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -183,10 +201,45 @@ func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		return makeID(r.c.TrustDomain, r.getIdentityTemplate(pod))
+	}
+
+	// the controller has not been configured with a specific identity format.
 	// create an entry based on the service account.
 	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
 }
 
 func (r *PodReconciler) podParentID(nodeName string) string {
 	return makeID(r.c.TrustDomain, "k8s-workload-registrar/%s/node/%s", r.c.Cluster, nodeName)
 }
+
+func (r *PodReconciler) getIdentityTemplate(pod *corev1.Pod) string {
+	tpl := r.c.IdentityTemplate
+	tmpl, err := template.New("IdentityTemplate").Parse(tpl)
+	if err != nil {
+		r.c.Log.WithError(err).Errorf("Error parsing the template (%v)", tpl)
+	}
+
+	// Create the IdentityMaps struct, with maps, one for Pod and one for Context:
+	podInfo := map[string]string{
+		PodServiceAccountIDLabel: pod.Spec.ServiceAccountName,
+		NamespaceIDLabel:         pod.Namespace,
+		PodNameIDLabel:           pod.Name,
+		PodUIDLabel:              string(pod.UID),
+		PodHostnameLabel:         pod.Spec.Hostname,
+		PodNodeNameLabel:         pod.Spec.NodeName,
+	}
+
+	templateMaps := IdentityMaps{
+		Context: r.c.Context,
+		Pod:     podInfo,
+	}
+	var svid bytes.Buffer
+	err = tmpl.Execute(&svid, templateMaps)
+	if err != nil {
```

Comment:
in case no error is returned what happens in the client of this function that will get an empty svid?

Reply:
Empty SVID means `ignore`. I fixed the error flow.